packages/react-dev-overlay/src/internal/container/BuildError.tsx

```diff
@@ -61,4 +67,97 @@ export const styles = css`
   .nextjs-container-build-error-body small {
     color: #757575;
   }
+
+  .nextjs-container-build-error-version-status {
+    font-weight: bold;
+  }
+  .nextjs-container-build-error-version-status span {
+    display: inline-block;
+    width: 10px;
+    height: 10px;
+    border-radius: 5px;
+    background: #eaeaea;
+    margin-right: 6px;
+  }
+
+  .nextjs-container-build-error-version-status span.fresh {
+    background: #50e3c2;
+  }
+
+  .nextjs-container-build-error-version-status span.outdated {
+    background: #e00;
+  }
+
+  .nextjs-container-build-error-version-status span.stale {
+    background: #f5a623;
+  }
 `
+declare global {
+  interface Window {
+    next: { version: string; latest: string }
+  }
+}
+
+type Staleness = 'fresh' | 'stale' | 'outdated'
+
+const NextStatus = () => {
+  const [staleness, setStaleness] = React.useState<Staleness>()
+  React.useEffect(() => {
+    ;(async () => {
+      try {
+        const installedVersion = window.next.version.replace(/-canary.*/, '')
+
+        const latestVersion = await getLatest()
+        console.log(latestVersion, installedVersion)
+
+        const result = compareVersions(installedVersion, latestVersion)
+        setStaleness(result)
+      } catch {}
+    })()
+  }, [])
+
+  const title = {
+    fresh: 'Next.js is up to date!',
+    stale: 'Next.js is out of date, update recommended!',
+    outdated: 'Next.js is out of date, update necessary!',
+    undetermined: 'Could not determine Next.js version status.',
+  }[staleness ?? 'undetermined']
+
+  return (
+    <div title={title} className="nextjs-container-build-error-version-status">
+      <span className={staleness} />
+      Next.js {window.next.version}
+    </div>
+  )
+}
+
+async function getLatest() {
+  return '12.1.6'
```

Comment:
Should be removed? Or a different logic on the server-side should be implemented instead?

Reply:
Thanks, just a WIP :+1: I hit a rate limit on the API below.